packages/transition-backend/src/services/executableJob/ExecutableJob.ts

```diff
@@ -50,6 +51,37 @@ export class ExecutableJob<TData extends JobDataType> extends Job<TData> {
         }
     }
 
+    /**
+     * Prepare the input file for a job based on the requested file location.
+     * The result of this function can be passed to the `createJob` function as
+     * an input file.
+     *
+     * @param fileLocation The file location configuration, either in the upload
+     * directory or from another job
+     * @param userId The user ID requesting the File
+     * @param importFileName The name of the file to use if the file is from an
+     * upload FIXME this should probably be in the file config already
+     * @returns
+     */
+    static handleJobFile = async (
+        fileLocation: FileConfig,
+        userId: number,
+        importFileName: string
+    ): Promise<string | { filepath: string; renameTo: string }> => {
+        if (fileLocation.location === 'upload') {
+            return {
+                filepath: `${directoryManager.userDataDirectory}/${userId}/imports/${importFileName}`,
```

Comment:
Since it does not operate on a job directory, I don't think this should be part of the ExecutableJob, 

ExecutableJob should just get a path at this point and not be aware of a specific imports directory.